src/datasource.ts

```diff
@@ -24,4 +30,53 @@ export class DataSource extends DataSourceWithBackend<AthenaQuery, AthenaDataSou
 
   applyTemplateVariables = (query: AthenaQuery, scopedVars: ScopedVars) =>
     applySQLTemplateVariables(query, scopedVars, getTemplateSrv);
+
+  getVariables = () => this.templateSrv.getVariables().map((v) => `$${v.name}`);
+
+  getRegions = () => this.getResource('regions').then((regions) => appendTemplateVariables(this, regions));
+
+  getCatalogs = (query: AthenaQuery) =>
+    this.postResource('catalogs', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+    }).then((catalogs) => appendTemplateVariables(this, catalogs));
+
+  getDatabases = (query: AthenaQuery) =>
+    this.postResource('databases', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+      catalog: this.templateSrv.replace(query.connectionArgs.catalog),
+    }).then((databases) => appendTemplateVariables(this, databases));
+
+  getTables = (query: AthenaQuery) =>
+    this.postResource('tables', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+      catalog: this.templateSrv.replace(query.connectionArgs.catalog),
+      database: this.templateSrv.replace(query.connectionArgs.database),
+    }).then((tables) => appendTemplateVariables(this, tables));
+
+  getColumns = (query: AthenaQuery) =>
+    this.postResource('columns', {
+      region: this.templateSrv.replace(query.connectionArgs.region),
+      catalog: this.templateSrv.replace(query.connectionArgs.catalog),
+      database: this.templateSrv.replace(query.connectionArgs.database),
+      table: this.templateSrv.replace(query.table),
+    }).then((columns) => appendTemplateVariables(this, columns));
+
+  query(options: DataQueryRequest<AthenaQuery>): Observable<DataQueryResponse> {
+    options = cloneDeep(options);
+
+    const queries = options.targets.filter((item) => item.hide !== true);
```

Comment:
I am not sure if the default case (I would expect so), maybe it's not because we are overwritting `filterQuery`. Can you check it? In any case, that condition should be in that function.